dev_docs/getting_started/setting_up_a_development_env.mdx

```diff
@@ -92,3 +92,28 @@ node scripts/register_git_hook
 ```
 
 After the script completes the pre-commit hook will be created within the file `.git/hooks/pre-commit`. If you choose to not install it, don’t worry, we still run a quick CI check to provide feedback earliest as we can about the same checks.
+
+## Using the Kibana Dev Container (optional)
+
+Kibana also supports using a [dev container](https://containers.dev/) which can integrate with various editors and tools [(supported tools)](https://containers.dev/supporting). The dev container provides a consistent development environment across different machines and setups which is based on Ubuntu Jammy (22.04). The only prerequisite is having [Docker](https://www.docker.com/) installed locally. VS Code is the recommended editor and will be used for these instructions because it is the most mature, but it is not required.
+
+### Setting up the Dev Container
```

Comment:
I'm guessing functional tests won't be supported inside the container.  Thoughts on adding a limitations section?

Any ideas if debugging works?

Reply:
I ran some headless functional tests in the dev container for both stateful and serverless, and they worked fine, but it wasn't extensive testing. Do you have some configs in mind?

I forwarded the debugging the ports and it is working through Chrome:

```bash
# kbn logs
Debugger attached.
Debugger ending on ws://127.0.0.1:9229/0579f46f-c9b4-4e8b-b61a-8f80f7cab393
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Debugger ending on ws://127.0.0.1:9229/0579f46f-c9b4-4e8b-b61a-8f80f7cab393
```

Reply:
Mostly not headless, I expect developers adding tests are going to want to see the browser.

Not worried about it, I'll leave it up to you.

Reply:
I added 300b325d2a26d1c30ccb5e0ccae63ab6fc2837c7 for headed browser support. This was working on my machine but I already had setup things to pipe GUI applications from WSL to Windows. I think the devcontainer is separated from that and doesn't seem to be interacting with it, but it would probably be good for someone else to give it a shot.

`node scripts/functional_tests --config=test/ui_capabilities/newsfeed_err/config.ts` was the command I was running in the devcontainer to run tests which utilize Chrome.